types/test/model.ts

```diff
@@ -1,6 +1,6 @@
 import { Association, DataTypes, HasOne, Model, Sequelize } from 'sequelize';
 
-class MyModel extends Model {
+class MyModel extends Model<MyModel> {
```

Comment:
Is there a reason for this? Those `T`s in `Model<T = any, T2 = any>` doesn't seem to be used.

Reply:
@GuskiS indeed they are not used, they are going to be removed, see #11682

Reply:
The first generic type is used with this PR in order to get autocompletion for `new MyModel(/* here */)`

Reply:
@RobinBuschmann ah, so this means that [I'm wrong here](https://github.com/sequelize/sequelize/pull/11682#issuecomment-557054719)? good to know, please mention it on that PR so we don't forget about that, thanks

Reply:
@RobinBuschmann Oh, I see. For me, it is very confusing that it takes class that is being created on the same line.

Reply:
@GuskiS I know what you mean, but unfortunately it isn't possible to use `this` for the constructor params (`TS2526:  A 'this' type is available only in a non-static member of a class or interface.`), which makes sense, since the type of `this` isn't determined at this point.

Reply:
I wonder, can't we do something like this? https://www.typescriptlang.org/play/#code/JYOwLgpgTgZghgYwgAgLIE9UHsAmEA2AgmGFMAEYCukAzsgN4BQyywOAXMiJQLbnQBuZskgAPMJxqlQAcyEBfRowT44NOgCE1EbHnwAeACoA+BksXLV6tJlwFkEcRBA5N23QX0YPREmSq0psA8AA74EDzOYHTedr7SARB0TIzyQA

This way you won't have to remove the model stuff from the type.

Reply:
Hey @GuskiS, yes this is an option. But then you would need to define interfaces for every model. For each interface it is very likely that it is only implemented by one class and you would have to define the properties one more time on the model itself. So inferring the value types of the model class itself is much more comfortable for the consumer of this library. And in addition nothing prevents you from defining an interface anyway. What do you think?

Reply:
Yea, that makes sense.
However, in my app we still create that interface because we are using it to define express server's request body type. Using the Model type in this is would be incorrect.

Anyways, that was just a thought and an example :) Waiting for this get merged 👼

Reply:
Perhaps both are achievable? Default the attributes to `ModelValues<M>`, but allow them to be overridden using the pattern @GuskiS is proposing?

Reply:
Perhaps both are achievable? Default the attributes to ModelValues<M>, but allow them to be overridden using the pattern @GuskiS is proposing?